Sources/Deprecations.swift

```diff
@@ -41,3 +41,46 @@ public extension Promise {
         return ensure(on: q, body)
     }
 }
+
+public extension Thenable {
+    @available(*, deprecated: 6.1, message: "See: `compactMap`")
+    func flatMap<U>(on: DispatchQueue? = conf.Q.map, _ transform: @escaping(T) throws -> U?) -> Promise<U> {
+        return compactMap(on: on, transform)
+    }
+}
+
+public extension Thenable where T: Sequence {
+    @available(*, deprecated: 6.1, message: "See: `mapValues`")
+    func map<U>(on: DispatchQueue? = conf.Q.map, _ transform: @escaping(T.Iterator.Element) throws -> U) -> Promise<[U]> {
+        return mapValues(on: on, transform)
+    }
+
+    @available(*, deprecated: 6.1, message: "See: `flatMapValues`")
+    func flatMap<U: Sequence>(on: DispatchQueue? = conf.Q.map, _ transform: @escaping(T.Iterator.Element) throws -> U) -> Promise<[U.Iterator.Element]> {
+        return flatMapValues(on: on, transform)
+    }
+
+    @available(*, deprecated: 6.1, message: "See: `filterValues`")
+    func filter(on: DispatchQueue? = conf.Q.map, test: @escaping (T.Iterator.Element) -> Bool) -> Promise<[T.Iterator.Element]> {
+        return filterValues(on: on, test)
+    }
+}
+
+public extension Thenable where T: Collection {
+    @available(*, deprecated: 6.1, message: "See: `firstValue`")
+    var first: Promise<T.Iterator.Element> {
+        fatalError()
```

Comment:
Should this be `return firstValue`? You have `last` implemented as `return lastValue`.

Reply:
Good spot